initrd/init

```diff
@@ -337,19 +294,62 @@ for c in `cat /proc/cmdline`; do
 done
 
 cd /
-while :; do
-	for device in ${ROOT:-/dev/[hmnsvx][dmrv][0-9a-z]*}; do
-		check_root $device && break 2
+for device in /dev/[hmnsvx][dmrv][0-9a-z]*; do
+	ln $device /dev/block
+done
 
-		# Also umount /android and /sfs if already mounted
-		mountpoint -q /android && umount /android
-		mountpoint -q /sfs && umount /sfs
-		mountpoint -q /mnt && umount /mnt
-	done
-	sleep 1
-	echo -n .
+# if the android is partition table-based
+IS_ALL_BLOCKDEV=1
+for dev in $SYSTEM $DATA $RECOVERY; do
```

Comment:
recovery is a gzip cpio image just like initrd.img so there's no `$RECOVERY` option

forgot to tell you that

Reply:
I already know
This is for cases like partition table-based installation so either use another `init` or create some other variables works